migrations/helpers/update_legacy_filtered_positives_helper.py

```diff
@@ -0,0 +1,191 @@
+                f" JOIN {events_wh_db}.subjects mlwh_events_subjects ON (mlwh_events_subjects.friendly_name = sanger_sample_id)"  # noqa: E501
+                f" JOIN {events_wh_db}.roles mlwh_events_roles ON (mlwh_events_roles.subject_id = mlwh_events_subjects.id)"  # noqa: E501
+                f" JOIN {events_wh_db}.events mlwh_events_events ON (mlwh_events_roles.event_id = mlwh_events_events.id)"  # noqa: E501
+                f" JOIN {events_wh_db}.event_types mlwh_events_event_types ON (mlwh_events_events.event_type_id = mlwh_events_event_types.id)"  # noqa: E501
+                f" WHERE mlwh_sample.description IN %(root_sample_ids)s"
+                f" AND mlwh_sample.created BETWEEN '{start_date}' AND '{end_date}'"
+                f" AND mlwh_stock_resource.labware_human_barcode IN %(plate_barcodes)s"
+                " AND mlwh_events_event_types.key = 'cherrypick_layout_set'"
+                " GROUP BY mlwh_sample.description, mlwh_stock_resource.labware_human_barcode, mlwh_sample.phenotype, mlwh_stock_resource.labware_coordinate"  # noqa: E501
+            )
+
+            frame = pd.read_sql(
+                sql,
+                db_connection,
+                params={
+                    "root_sample_ids": tuple(chunk_root_sample_id),
+                    "plate_barcodes": tuple(plate_barcodes),
+                },
+            )
+
+            # drop_duplicates is needed because the same 'root sample id' could
+            # pop up in two different batches, and then it would
+            # retrieve the same rows for that root sample id twice
+            # do reset_index after dropping duplicates to make sure the rows are numbered
+            # in a way that makes sense
+            concat_frame = concat_frame.append(frame).drop_duplicates().reset_index(drop=True)
+
+        return concat_frame
+    except Exception as e:
+        logger.error("Error while connecting to MySQL")
+        logger.exception(e)
+        return None
+    finally:
+        db_connection.close()
+
+
+def v0_version_set(config: ModuleType):
+    """Find if the v0 or v1 version has been set in any of the samples.
+       This would indicate that the legacy migration has already been run.
+
+    Args:
+        samples {List[Dict]} -- List of samples from Mongo
+
+    Returns:
+        Boolean -- v0 version set in samples
+    """
+    with create_mongo_client(config) as client:
+        mongo_db = get_mongo_db(config, client)
+        samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
+
+        v0_samples = list(
+            samples_collection.find({FIELD_FILTERED_POSITIVE_VERSION: FILTERED_POSITIVE_VERSION_0})
+        )  # noqa: E501
+        if len(v0_samples) > 0:
+            return True
+        else:
+            return False
+
+
+def split_mongo_samples_by_version(
+    samples: List[Sample], cp_samples_df_v0: DataFrame, cp_samples_df_v1: DataFrame
+):  # noqa: E501
```

Comment:
Can specify the return type here, and update description. I think the type should be `Dict[FilteredPositiveIdentifier, List[Sample]]`